src/AutoMapper/Execution/ExpressionBuilder.cs

```diff
@@ -62,29 +64,29 @@ Expression ClearDestinationCollection()
             {
                 var destinationElementType = ElementTypeHelper.GetElementType(destinationParameter.Type);
                 var destinationCollectionType = typeof(ICollection<>).MakeGenericType(destinationElementType);
-                var destinationVariable = Expression.Variable(destinationCollectionType, "collectionDestination");
+                var destinationVariable = Variable(destinationCollectionType, "collectionDestination");
                 var clearMethod = destinationCollectionType.GetDeclaredMethod("Clear");
-                var clear = Expression.Condition(Expression.Property(destinationVariable, "IsReadOnly"),
-                    Expression.Empty(), Expression.Call(destinationVariable, clearMethod));
-                return Expression.Block(new[] {destinationVariable},
-                    Expression.Assign(destinationVariable,
+                var clear = Condition(Property(destinationVariable, "IsReadOnly"),
```

Comment:
I would suggest for static calls with 1-2 maybe 3 variables use Extension methods like in my cleanup/optimize [PR](https://github.com/AutoMapper/AutoMapper/pull/1346/files#diff-d58ee48fd6246bad6783dc201ff089aaR68)

Then like will look like
```
var clear = destinationVariable.Property("IsReadOnly").Condition(Empty(), destinationVariable.Call(clearMethod))
```
And hopefully that fluent API will actually be easier to read when people look back at this code.